core/schema_utils.py

```diff
@@ -134,7 +136,18 @@ def normalize_against_schema(
                 prop[SCHEMA_KEY_SCHEMA],
                 global_validators=global_validators
             )
+    elif schema[SCHEMA_KEY_TYPE] == SCHEMA_TYPE_DICT_WITH_VARIABLE_NO_OF_KEYS:
+        assert isinstance(obj, dict), ('Expected dict, received %s' % obj)
+        schema_value_type = schema[SCHEMA_KEY_VALUES]
+        normalized_obj = {}
+        for key, value in obj.items():
+            normalized_obj[key] = normalize_against_schema(
+                value, schema_value_type[SCHEMA_KEY_SCHEMA],
+                global_validators=global_validators
+            )
```

Comment:
This looks good, should we allow/require a check of the type of the keys? Something similar to `SCHEMA_KEY_LEN` in the `SCHEMA_TYPE_LIST`?

Reply:
Also add tests for this.

Reply:
I have added a condition to check if the key is of type `str`.